### ISSUE TITLE
Detect errors in hpcflow internal calls

### ISSUE THREAD

**Issue by dkfellows** [hpcflow/hpcflow-new#714]
If an error happens during, say, the writing of the EAR end timestamp, the overall workflow ends up in a very strange state, thinking that the EAR succeeded and yet failed at the same time (observed on Windows at a minimum). At the very least, a failure during this point ought to bubble out and cause the whole jobscript to fail (even better would be if the exception trace ended up in the main log for the workflow, but that might be harder to achieve).

Note that this is definitely a "shouldn't happen" case that only happens during development, but it took me ages to realise what was going wrong when I hit it (during #698).

**Comment by aplowman** [comment]
Yeah, it can get confusing. This problem should be improved somewhat in the `refactor/jobscript` branch, because I have made an attempt to separate out standard output/error streams that originate from the software from those that originate from hpcflow. So it's a bit easier to see what happened.
